### PR TITLE
Task dev 7 inh

### PR DESCRIPTION
**Pull request by ineskavalenka**

### CODE REVIEW

File: EquilateralTriangle.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import java.math.BigDecimal;
+
+/**
+ * Describes equilateral triangles.
+ */
+class EquilateralTriangle extends Triangle {
+  double repeatingSide;
+  private String type = new String("equilateral");
```
Comment (AlexanderSoroka) [inline EquilateralTriangle.java]: it must be const

File: EquilateralTriangle.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import java.math.BigDecimal;
+
+/**
+ * Describes equilateral triangles.
+ */
+class EquilateralTriangle extends Triangle {
+  double repeatingSide;
```
Comment (AlexanderSoroka) [inline EquilateralTriangle.java]: be simple - just `side` is enought

File: EquilateralTriangle.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+   * Determines the existence of an equilateral triangle with given sides.
+   *
+   * @return true if the triangle exists and is equilateral,
+   *         false if not.
+   */
+  public boolean exists(double[] sides) {
+    if (!super.exists(sides)) {
+      return false;
+    }  
+    BigDecimal sideA = new BigDecimal(sides[0]);
```
Comment (AlexanderSoroka) [inline EquilateralTriangle.java]: avoid that cast from double to BigDecimal - let sides be BigDecimal from very beginning

File: EquilateralTriangle.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    BigDecimal sideB = new BigDecimal(sides[1]);
+    BigDecimal sideC = new BigDecimal(sides[2]);
+    if (sideA.equals(sideB) && sideA.equals(sideC)) {
```
Comment (AlexanderSoroka) [inline EquilateralTriangle.java]: return sideA.equals(sideB) && sideA.equals(sideC);

File: InputSequenceProvider.java [hunk 5]
```diff
@@ -22,8 +22,30 @@ public InputSequenceProvider(String[] args) {
...
+  /**
+   * Gets the next input triangle sequence (a sequence of triangle sides).
+   * When called for the first time, returns the value of firstSequence field as a sequence,
+   * and sets that field empty.
+   * When the subsequent callings occur, requests sequences from the console. 
+   * 
+   * @return the sequence.
+   */ 
+  public String[] provideInputTriangle() {
+    // to do : how should I do this not violating DRY ?
```
Comment (AlexanderSoroka) [inline InputSequenceProvider.java:40]: what do you mean DRY?

File: IsoscelesTriangle.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import java.math.BigDecimal;
+
+/**
+ * Describes isosceles triangles.
+ */
+class IsoscelesTriangle extends Triangle {
+  double singleSide;
```
Comment (AlexanderSoroka) [inline IsoscelesTriangle.java]: BigDecimal

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: InputSequenceProvider.java (hunk 5 above)
Comment (ineskavalenka) [reply InputSequenceProvider.java:40]: "Do not Repeat Yourself"
this method is almost a Ctrl+C / Ctrl+V of the provideInputSequence()

File: WhatTriangle.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        System.out.println("The triangle is " + equilateral.build(sides).toString() + ".");
+        return;
+      } catch (NumberFormatException e) {
+        System.out.println("Incorrect input arguments.");
+        System.out.println("The sequence of sides should contain three positive numbers.");  
+      } catch (IllegalArgumentException e) {
+        System.out.println("A triangle must have 3 sides.");
+      } catch (NullPointerException e) { 
+        System.out.println("The triangle does not exist (S <= 0).");
+      }
```
Comment (ineskavalenka) [inline WhatTriangle.java]: I expected it to separate "not three triangle sides exception" behavior from the "geometrical non-existence exception" behavior, but it does not work, and I have no idea how to make it work.

Are they generally different situations? Kheidorov told me they all are "incorrect input".

**Pull request closed without merging**